app/src/main/java/com/eveningoutpost/dexdrip/Models/Treatments.java

```diff
@@ -75,42 +79,83 @@
     @Column(name = "carbs")
     public double carbs;
     @Expose
-    @Column(name = "insulin")
```

Comment:
Why remove/change the name of this field considering the large number of code changes and incompatibilities this introduces and invalidating historical data? Some people have years of data stored.  Is there some special reason for this?

Reply:
the main reason was to catch all lines of code "doing something" with this insulin dose.
As the new prediction logic can't use these "old data" they seemed to be useless. BUT if necessary we can write an alter script to transform all the old data to the new structure.

Just for understanding (we use xdrip since 6 months and write all data to nightscout to get our reportings): 
why are those people storing their data in xdrip instead of tidepool or NS?

Reply:
Can you explain what the difference is between the `insulinSummary` field vs the `insulin` field if the user is only using a single insulin type of for example NovaRapid?

Reply:
insulinSummary is just the sum of all injection doses in a treatment. Whenever a logic needs to have the injected dose (e.g. for sending to tidepool or NS or just to calculate or report something even within nightscout or any other consumption system) it has the insulin dose to do his work without the differentiation of profiles.
When a user is using just one profile of insulin (e.g. pump without basal), insulinSummary is the same as before. But in any other therapy (CT, FIT, 2/3-Pen-Scheme...) there is the difference as expected.....

Reply:
Okay, that is what I thought and I wanted to check that. Thanks. 

I will look at a way to get this integrated that causes the minimum impact to the existing behavior that gives the users the choice to use multiple insulin types.

Reply:
from my point of view this renaming in the db schema is solvable by implementing an alter script in the treatment class (there is still an existing one but i haven't used it). This will reestablish compatibility INSIDE xdrip+. Whenever someone is using this db outside of xdrip+ app there will be issues too (but i can't imagine how to implement such a usage outside of xdrip6)

Reply:
That wouldn't solve the issue when users downgrade xDrip versions. There is schema compatibility right back to 2016 versions. I don't see a reason that the name needs to be changed  to start with.

Reply:
downgrading xdrip? As far as i can see there is not schema downgrad procedure implemented....
BUT to make clear: this insulinSummary is not the same as the "old" insulin attribute - both have different meanings because of the changed behaviour.....